src/SFA.DAS.Tools.Servicebus.Support.Infrastructure/Services/ICosmosDbContext.cs

```diff
@@ -16,5 +16,9 @@ public interface ICosmosDbContext
         Task<QueueMessage> GetQueueMessageAsync(string userId, string messageId);
         Task<int> GetMessageCountAsync(string userId, SearchProperties searchProperties = null);
         Task<bool> HasUserAnExistingSession(string userId);
+        Task<UserSession> UpsertUserSessionAsync(UserSession userSession);
+        Task<UserSession> GetUserSessionAsync(string userId);
+        Task DeleteUserSessionAsync(string id, string userId);
+        Task<IEnumerable<UserSession>> GetExpiredUserSessions();
```

Comment:
These methods seem different enough to be in their own separate interface?

Reply:
@DannyGB refactored